btx/processing/rawimagetimetool.py

```diff
@@ -0,0 +1,575 @@
+            self.plot_hist(edges)
+
+    def plot_calib(self, delays: list, edges: list, model: list):
+        """! Plot the density of detected edges during a time tool calibration
+        run and the polynomial fit to it.
+
+        @param delays (array-like) Time tool delays in nanoseconds.
+        @param edges (array-like) Detected edges in time tool camera.
+        @param model (array-like) Polynomial model coefficients.
+        """
+        poly = np.zeros([len(edges)])
+        n = len(model)
+        for i, coeff in enumerate(model):
+            poly += coeff*edges**(n-i-1)
+
+        fig, ax = plt.subplots(1, 1)
+        ax.hexbin(edges, delays, gridsize=50, vmax=500)
+        ax.plot(edges, poly, 'ko', markersize=2)
+        ax.set_xlabel('Edge Pixel')
+        ax.set_ylabel('Delay')
+        run = self.format_run()
+        fname = f'TTCalib_{run}.png'
+        self.write_file(fig, fname, f'{self.savedir}/figs')
+
+    def plot_hist(self, edges: list):
+        """! Plot the density of detected edges in a histogram. For calibration
+        runs this should be uniform. For experiment runs this will ideally be
+        quasi-Gaussian.
+
+        @param edges (array-like) Detected edges in time tool camera.
+        """
+        fig, ax = plt.subplots(1, 1)
+        ax.hist(edges, bins=20, density=True)
+        ax.set_xlabel('Edge Pixel')
+        ax.set_ylabel('Density')
+        run = self.format_run()
+        fname = f'EdgeHist_{run}.png'
+        self.write_file(fig, fname, f'{self.savedir}/figs')
+
+    def write_file(self, fobj, fname: str, savedir: str, fmt = None):
+        """! Writes objects of multiple types to output files. Checks if the
+        save directory exists, if not it creates it.
+
+        @param fobj (plt.Figure | np.array) Object to write to disk. Can be a figure or array.
+        @param fname (str) Output filename.
+        @param savedir (str) Save directory.
+        """
+        try:
+            os.makedirs(f'{savedir}')
+        except FileExistsError:
+            pass
+        finally:
+            if type(fobj) == plt.Figure:
+                fobj.savefig(f'{savedir}/{fname}')
+            elif type(fobj) == np.ndarray:
+                if fmt:
+                    np.savetxt(f'{savedir}/{fname}', fobj, fmt)
+                else:
+                    np.savetxt(f'{savedir}/{fname}', fobj)
+
+    #@todo Implement to select individual images
+    def get_images(ds: psana.DataSource, det: psana.Detector) -> (list):
+        pass
+
+   # if direct:
+   #     for idx,evt in enumerate(ds.events()):
+   #         ec = evr_det.eventCodes(evt)
+   #         if ec is None: continue
+   #         edge_pos = np.append(edge_pos,ds.env().epicsStore().value(f'{beamline}:TIMETOOL:FLTPOS'))
+   #         edge_fwhm = np.append(edge_fwhm,ds.env().epicsStore().value(f'{beamline}:TIMETOOL:FLTPOSFWHM'))
+   #         edge_amp = np.append(edge_amp,ds.env().epicsStore().value(f'{beamline}:TIMETOOL:AMPL'))
```

Comment:
remove?

Reply:
Can be removed.

This was a note for if we want to implement a function for the scenario where we just trust the timetool output without rerunning it. These are the EPICS PVs we can look at to access them.